docs/source/fx.rst

```diff
@@ -7,250 +7,181 @@ Overview
-
 Writing Transformations
 -----------------------
 
-TODO
-
-
-Debugging
------------
-
-Introduction
-^^^^^^^^^^^^^^^^
-
-After symbolically tracing an ``nn.Module`` and performing some number
-of transformations on the resulting GraphModule, we'll want to verify
-that the proper semantics were preserved after those transforms. If they
-weren't, we may need to do some debugging. The key is to work
-backwards: first, check the results of the generated module, then debug
-the generated code, then debug the process of transformations that lead
-to the generated code.
-
-If you’re not familiar with debuggers, please see the auxiliary section
-:ref:`Available Debuggers`.
-
-Debugging the Generated Code
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-Because FX generates the ``forward()`` function on GraphModules, using
-traditional debugging techniques like ``print`` statements or ``pdb`` is
-not as straightfoward. Luckily, we have several techniques we can use
-for debugging the generated code.
-
-Use ``pdb``
-~~~~~~~~~~~~~
-Invoke ``pdb`` to step into the running program. Although the code that
-represents the FX graph is not in any source file, we can still step
-into it manually using ``pdb`` when the forward pass is invoked.
+What is a FX transform? Essentially, it's a function that looks like this.
 
 ::
 
-    def my_pass(in: torch.nn.Module) -> torch.nn.Module:
-        traced = torch.fx.symbolic_trace(in)
-        # Transformation logic here
-        return traced
+    def transform(m: nn.Module) -> nn.Module:
+        fx_model: GraphModule = fx.symbolice_trace(m)
+        new_model = ...
+        return new_model
 
-    # When this line is executed at runtime, we will be dropped into an
-    # interactive `pdb` prompt. We can use the `step` or `s` command to
-    # step into the execution of the next line
-    import pdb; pdb.set_trace()
+Your transform will take in an :class:`torch.nn.Module`, convert it into a
+:class:`GraphModule` with :meth:``symbolic_trace``, and return a new
+``nn.Module``. You should think of the ``nn.Module`` that your FX transform
+returns as identical to a regular ``nn.Module`` -- you can pass it to another
+FX transform, you can pass it to TorchScript, or you can
+run it. Ensuring that the inputs and outputs of your FX transform are a
+``nn.Module`` will allow for composability.
 
-    my_pass(my_module)
+Given that you’ve passed in an ``nn.Module`` that has been traced into a
+graph, there are now two primary approaches you can take to building a new
+graph.
 
-.. _Print the Generated Code:
+Graph Manipulation
+^^^^^^^^^^^^^^^^^^
 
-Print the Generated Code
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
-If you’d like to run the same code multiple times, then it can be
-a bit tedious to step to the right code with ``pdb``. In that case, one
-approach is to simply copy-paste the generated ``forward`` pass into
-your code and examine it from there.
+One approach to building this new graph is to simply transform your old
```

Comment:
lol `simply transform` sounds like a tautology here. (How do you transform the graph? just transform it!) Maybe say "One approach to building this new graph is to directly maniulate your old one"